src/mqt/bench/benchmark_generation.py

```diff
@@ -55,30 +55,33 @@ def generate_filename(
     benchmark_name: str,
     level: str,
     num_qubits: int | None,
+    is_mirror: bool = False,  # New parameter
     target: Target | None = None,
     opt_level: int | None = None,
 ) -> str:
     """Generate a benchmark filename based on the abstraction level and context.
 
     Arguments:
         benchmark_name: name of the quantum circuit
-        level: abstraction level
+        level: original abstraction level ("alg", "indep", "nativegates", "mapped")
         num_qubits: number of qubits in the benchmark circuit
+        is_mirror: whether a mirror circuit is being generated
         target: target device (used for 'nativegates' and 'mapped')
-        opt_level: optional optimization level (used for 'nativegates' and 'mapped')
+        opt_level: optional optimization level (used for 'indep', 'nativegates', and 'mapped')
 
     Returns:
         A string representing a filename (excluding extension) that encodes
         all relevant metadata for reproducibility and clarity.
     """
-    base = f"{benchmark_name}_{level}"
+    level_str_modified = f"{level}_mirror" if is_mirror else level
+    base = f"{benchmark_name}_{level_str_modified}"
```

Comment:
Would have probably written this as something like (untested)
```suggestion
    base = f"{benchmark_name}_{level}{'_mirror' if generate_mirror_circuit}"
```
Highly likely that the above is not yet syntactically correct, but you get the idea